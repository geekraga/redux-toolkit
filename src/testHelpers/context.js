import { PropTypes } from 'react';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { IntlProvider, intlShape } from 'react-intl';

// takes array of message keys and returns an object where keys and values are the same
function buildMessages(messageKeys = []) {
    return messageKeys.reduce((messages, key) => ({ ...messages, [key]: key }), {});
}

export function getIntl(messageKeys = []) {
    const intlProvider = new IntlProvider({ locale: 'en', messages: buildMessages(messageKeys) }, {});
    return intlProvider.getChildContext().intl;
}

// returns context and childContextTypes for intl and muiTheme
export function standardContext({ messageKeys = [] } = {}) {
    return {
        context: { muiTheme: getMuiTheme(), intl: getIntl(messageKeys) },
        childContextTypes: { muiTheme: PropTypes.object, intl: intlShape },
    };
}
