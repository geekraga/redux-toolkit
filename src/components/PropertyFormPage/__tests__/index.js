/* eslint-env jest */

import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { getIntl } from '../../../testHelpers/context';
import SelectPropertyComponent from '../selectPropertyVideos';

describe('SelectPropertyComponent', () => {
    let component;

    beforeEach(() => {
        const messageKeys = ['cancel'];
        const intlComp = getIntl(messageKeys);
        const baseProps = {
            showLoader: false,
            fetchVideos: () => {},
            previewContent: () => {},
            propertyInfo: {},
            populateData: () => {},
            loadPropertyDetails: () => {},
            hideRecommendedVideos: () => {},
            uploadPropertyData: () => {},
            showRecommended: true,
            intl: intlComp,
        };
        const store = {
            default: () => {},
            subscribe: () => {},
            dispatch: () => {},
            getState: () => {
                return { ...state };
            },
        };
        component = shallow(<Provider store={store}><SelectPropertyComponent { ...baseProps } /></Provider>);
    });

    it('should render RecommendedVideos Component only', () => {
        expect(component).toMatchSnapshot();
    });
});

