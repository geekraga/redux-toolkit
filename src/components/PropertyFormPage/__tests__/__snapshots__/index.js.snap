// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPropertyComponent should render RecommendedVideos Component only 1`] = `
<SelectPropertyVideos
  categoryIdFilter={null}
  fetchVideos={[Function]}
  hideRecommendedVideos={[Function]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "cancel": "cancel",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  loadPropertyDetails={[Function]}
  populateData={[Function]}
  previewContent={[Function]}
  propertyInfo={Object {}}
  showLoader={false}
  showRecommended={true}
  uploadPropertyData={[Function]}
/>
`;
